cmake/Codegen.cmake

```diff
@@ -76,18 +76,23 @@ if(INTERN_BUILD_ATEN_OPS)
 
   file(GLOB_RECURSE all_python "${CMAKE_CURRENT_LIST_DIR}/../torchgen/*.py")
 
-  # RowwiseScaled.cu requires sm90a flags
+  # RowwiseScaled.cu requires sm89/sm90a flags
   if(USE_CUDA)
     set(ROWWISE_SCALED_MM_FILE "${CMAKE_CURRENT_LIST_DIR}/../aten/src/ATen/native/cuda/RowwiseScaledMM.cu")
 
     # Get existing arch flags
     torch_cuda_get_nvcc_gencode_flag(EXISTING_ARCH_FLAGS)
 
     # Check NVCC version and existing arch flags
-    if(CMAKE_CUDA_COMPILER_VERSION VERSION_GREATER_EQUAL 12.0 AND
-      EXISTING_ARCH_FLAGS MATCHES ".*compute_90.*")
-      set_source_files_properties(${ROWWISE_SCALED_MM_FILE}
-        PROPERTIES COMPILE_FLAGS "-gencode arch=compute_90a,code=sm_90a")
+    if(CMAKE_CUDA_COMPILER_VERSION VERSION_GREATER_EQUAL 12.0)
+      if(EXISTING_ARCH_FLAGS MATCHES ".*compute_86.*")
+        set_source_files_properties(${ROWWISE_SCALED_MM_FILE}
+          PROPERTIES COMPILE_FLAGS "-gencode arch=compute_89,code=sm_89")
+      endif()
+      if(EXISTING_ARCH_FLAGS MATCHES ".*compute_90.*")
+        set_source_files_properties(${ROWWISE_SCALED_MM_FILE}
+          PROPERTIES COMPILE_FLAGS "-gencode arch=compute_90a,code=sm_90a")
```

Comment:
I'm not great at reading cmake: does this mean this file will build for either-or sm90a/sm89, or would it build for both if 9.0 and 8.6 are in the arch list?

Reply:
It should build for both in that case.

Reply:
On a second look: it would not.  Fixed now.